workflow-core/src/main/java/com/squareup/workflow/Worker.kt

```diff
@@ -179,7 +178,7 @@ interface Worker<out OutputT> {
    * }
    * ```
    */
-  fun doesSameWorkAs(otherWorker: Worker<*>): Boolean = otherWorker::class == this::class
+  fun doesSameWorkAs(otherWorker: Worker<*>): Boolean = true
```

Comment:
Good point. The thinking was that the old default, comparing classes, would always be true now since the runtime itself is ensuring that worker types match. However, we ended up only comparing on the declared type of the worker, which means two workers that the runtime thinks are equivalent could still be different classes, so it would be safer to leave the class checks in here.

That said, by defaulting to relying on concrete types again, we lose the nice simplicity for testing. Not sure what the best thing to do is.

Reply:
Reverted this default implementation, and added a bunch of tests around it.

Reply:
Additional documentation added to `runningWorker` and `doesSameWorkAs`.